pkg/peers/peers_test.go

```diff
@@ -244,3 +251,433 @@ func TestMapNodesToPrimaryPodIPs(t *testing.T) {
+					Status: v1.PodStatus{PodIPs: []v1.PodIP{{IP: "10.0.2.2"}}},
+				},
+			},
+			expectedWorkerPeerCount:       1,
+			expectedControlPlanePeerCount: 2,
+			expectedWorkerIPs:             []string{"10.0.1.1"},
+			expectedControlPlaneIPs:       []string{"10.0.2.1", "10.0.2.2"},
+			description:                   "Should separate worker and control plane peers correctly",
+		},
+		{
+			name: "Control plane node with missing pod and empty IP",
+			nodes: []v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cp-1",
+						Labels: map[string]string{
+							hostnameLabelName:                       "cp-1",
+							"node-role.kubernetes.io/control-plane": "",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cp-2",
+						Labels: map[string]string{
+							hostnameLabelName:                       "cp-2",
+							"node-role.kubernetes.io/control-plane": "",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cp-3",
+						Labels: map[string]string{
+							hostnameLabelName:                       "cp-3",
+							"node-role.kubernetes.io/control-plane": "",
+						},
+					},
+				},
+			},
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "snr-pod-cp-1",
+						Labels: map[string]string{
+							"app.kubernetes.io/name":      "self-node-remediation",
+							"app.kubernetes.io/component": "agent",
+						},
+					},
+					Spec:   v1.PodSpec{NodeName: "cp-1"},
+					Status: v1.PodStatus{PodIPs: []v1.PodIP{{IP: "10.0.2.1"}}},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "snr-pod-cp-2",
+						Labels: map[string]string{
+							"app.kubernetes.io/name":      "self-node-remediation",
+							"app.kubernetes.io/component": "agent",
+						},
+					},
+					Spec:   v1.PodSpec{NodeName: "cp-2"},
+					Status: v1.PodStatus{PodIPs: []v1.PodIP{{IP: ""}}}, // Empty IP
+				},
+				// cp-3 has no pod
+			},
+			expectedWorkerPeerCount:       0,
+			expectedControlPlanePeerCount: 1,
+			expectedWorkerIPs:             []string{},
+			expectedControlPlaneIPs:       []string{"10.0.2.1"},
+			description:                   "Only cp-1 should be included (cp-2 has empty IP, cp-3 missing pod)",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a mock reader with the test data
+			mockReader := &mockReaderWithPeers{
+				nodes: tc.nodes,
+				pods:  tc.pods,
+			}
+
+			p := New(
+				"test-node",
+				50*time.Millisecond,
+				mockReader,
+				logr.Discard(),
+				5*time.Second,
+			)
+
+			// Run Start for a short time to allow peer updates
+			ctx, cancel := context.WithTimeout(context.Background(), 150*time.Millisecond)
+			defer cancel()
+
+			err := p.Start(ctx)
+			if err != nil {
+				t.Errorf("Start() returned an error when it should not: %v", err)
+			}
+
+			// Give it a moment to populate the addresses
+			time.Sleep(100 * time.Millisecond)
+
```

Comment:
It might be overkill, but the `require` package (https://pkg.go.dev/github.com/stretchr/testify/require) offers the `Eventually` assertion, much like in Ginkgo/Gomega, which might be useful here in place of `sleep`.